doc/source/serve/production.md

```diff
@@ -0,0 +1,651 @@
+      num_replicas: ...
+      ...
+    
+    - name:
+      ...
+    
+    ...
+```
+
+The `import_path` is the deployment graph's import path. When you deploy your config file, Serve will import your deployment graph using this path. Similarly, the `runtime_env` is the deployment graph's runtime environment. Serve will import the deployment graph inside this environment.
+
+The `deployments` section is optional. If it's omitted, Serve will launch the deployment graph (and all its deployments). The graph will run with any deployment settings specified in the `@serve.deployment` decorators from the graph's code. If you want to override these decorator settings from the code, you can include a `deployments` section in the file. You can add an entry of deployment settings to the `deployments` list. The only required setting in each list entry is the deployment `name`, which must match one of the deployments from the graph's code. You can include any settings from the `@serve.deployment` decorator inside the entry, **except** `init_args` and `init_kwargs`, which must be set in the graph's code itself.
+
+For example, let's take the `FruitStand` deployment graph from the [previous section](serve-in-production-testing). An equivalent config would be:
+
+```yaml
+import_path: fruit.deployment_graph
+
+runtime_env: {}
+
+deployments:
+
+    - name: FruitMarket
+      num_replicas: 2
+
+    - name: MangoStand
+      user_config:
+        price: 3
+    
+    - name: OrangeStand
+      user_config:
+        price: 2
+    
+    - name: PearStand
+      user_config:
+        price: 4
+    
+    - name: DAGDriver
+```
+
+The file uses the same `fruit.deployment_graph` import path, and it has five entries in the `deployments` list– one for each deployment. All the entries contain a `name` setting (the only required setting when including an entry) as well as additional settings (such as `num_replicas` or `user_config`) depending on the deployment.
+
+Note how this config specifies the same settings as the `@serve.deployment` decorators from the deployment graph's code. We can change or add to these settings to override the settings from the decorators.
+
+:::{tip}
+Each individual entry in the `deployments` list is optional. In the example config file above, we could omit the `PearStand`, including its `name` and `user_config`, and the file would still be valid. When we deploy the file, the `PearStand` deployment will still be deployed, using the configurations set in the `@serve.deployment` decorator from the deployment graph's code.
+:::
+
+We can also auto-generate this config file. The `serve build` command takes an import path to your deployment graph, and it creates a config file containing all the deployments and their settings from the graph. This file will also include all the default values set by Serve itself, so you can see exactly what configurations apply to your deployments. You can tweak these settings to manage you deployments in production.
```

Comment:
> it creates a config file containing all the deployments and their settings from the graph. This file will also include all the default values set by Serve itself,

I don't think we fully aligned on this yet and we should not document about this right now. Personally, I think we should not expose all the settings fields.

Reply:
Got it– I removed the sentence about the default values, but I left the rest of the paragraph in place. The description should be valid regardless of the direction we go in.